dom.go

```diff
@@ -454,6 +489,24 @@ func (tl *TokenList) Set(s []string) {
 	tl.SetString(strings.Join(s, " "))
 }
 
+type attribute struct {
+	key   string
+	value string
+}
+
+func (a attribute) Value() string {
+	return a.value
+}
+
+func (a attribute) Key() string {
+	return a.key
+}
+
+type Attribute interface {
```

Comment:
Indeed, after checking the documentation as pertaining to attributes, order is not generally a concerned, nor is it adhered to, also since data-set do return a map of DOMStrings, and since we cant support the ability to change values on the map and have it reflect with the DOM, I figured an struct is better. We can expose the attribute struct rather than use the Indirection interface.

Or would a map be a better fit?